Secret-agency-service/src/main/java/cz/fi/muni/pa165/secretagency/service/AgentService.java

```diff
@@ -0,0 +1,86 @@
+package cz.fi.muni.pa165.secretagency.service;
+
+import cz.fi.muni.pa165.secretagency.entity.Agent;
+import cz.fi.muni.pa165.secretagency.entity.Department;
+import cz.fi.muni.pa165.secretagency.entity.Mission;
+import cz.fi.muni.pa165.secretagency.enums.AgentRankEnum;
+import cz.fi.muni.pa165.secretagency.service.exceptions.AgentServiceException;
+
+import java.util.List;
+
+/**
+ * An interface that defines a service access to the {@link Agent} entity.
+ * @author (Adam Kral<433328>)
+ */
+interface AgentService {
+    /**
+     * Create new agent
+     * @param agent to be created
+     * @return created agent
+     */
+    Agent createAgent(Agent agent);
+
+    /**
+     * Get agent by his id
+     * @param id agent's id
+     * @return Agent with given id //TODO: throw exception on not found?
+     */
+    Agent getById(Long id);
```

Comment:
Nezůstaneme u toho, že tyhle obecné akce budou v GenericService, kterou budou ostatní  (konkrétnější) Service dědit?

Reply:
Udělal jsem na ní samostatný pull request s tím, že jsou u ní ošetřeny parametry a Dao se snažím injektovat přes Spring. Akorát nemám žádnou konkrétní Service, takže to není otestovaný. 😄 
Mohl bys to zkusit? Nejsem si jistý, jak si Spring poradím s tím, když má injektovat typový field.

Reply:
ja jsem to zkousel, ale nebyl jsem to schopen namockovat v testech 😢